functions/src/handlers/firestore_user.ts

```diff
@@ -4,18 +4,91 @@ import { firestore } from 'firebase-admin';
 
 export default class FirestoreUserHandler {
     private logger: Logger = new Logger('FirestoreUserHandler');
-    // private db: firestore.Firestore = firestore();
+    private db: firestore.Firestore = firestore();
 
     constructor() {
         this.logger.setLogLevel('debug');
     }
 
     public async newUserDocumentHandler(snapshot: firestore.QueryDocumentSnapshot, context: functions.EventContext) {
+        if (snapshot.data().role == 'nonpractitioner') {
+            try {
+                this.logger.info('The new user is identified by: ', snapshot.data().uid);
+                //set public profile
+                this.db.doc(`users/${snapshot.data().uid}/public_profile/${snapshot.data().uid}`).set({
+                    Name: snapshot.data().name,
```

Comment:
@wansom Could you also rename these variables to the corresponding variable names on the flutter models?

Reply:
i have,could you check the latest push?